bolt/lib/Core/BinarySection.cpp

```diff
@@ -148,37 +148,16 @@ void BinarySection::flushPendingRelocations(raw_pwrite_stream &OS,
 
 
   for (Relocation &Reloc : PendingRelocations) {
-    uint64_t Value = Reloc.Addend;
+    uint64_t TargetAddress = Reloc.Addend;
     if (Reloc.Symbol)
-      Value += Resolver(Reloc.Symbol);
-    switch(Reloc.Type) {
-    default:
-      LLVM_DEBUG(dbgs() << Reloc.Type << '\n';);
-      llvm_unreachable("unhandled relocation type");
-    case ELF::R_X86_64_64:
-    case ELF::R_X86_64_32: {
-      OS.pwrite(reinterpret_cast<const char*>(&Value),
-                Relocation::getSizeForType(Reloc.Type),
-                SectionFileOffset + Reloc.Offset);
-      break;
-    }
-    case ELF::R_X86_64_PC32: {
-      Value -= SectionAddress + Reloc.Offset;
-      OS.pwrite(reinterpret_cast<const char*>(&Value),
-                Relocation::getSizeForType(Reloc.Type),
-                SectionFileOffset + Reloc.Offset);
-      LLVM_DEBUG(dbgs() << "BOLT-DEBUG: writing value 0x"
-                        << Twine::utohexstr(Value) << " of size "
-                        << Relocation::getSizeForType(Reloc.Type)
-                        << " at offset 0x" << Twine::utohexstr(Reloc.Offset)
-                        << " address 0x"
-                        << Twine::utohexstr(SectionAddress + Reloc.Offset)
-                        << " Offset 0x"
-                        << Twine::utohexstr(SectionFileOffset + Reloc.Offset)
-                        << '\n';);
-      break;
-    }
-    }
+      TargetAddress += Resolver(Reloc.Symbol);
+
+    uint64_t Value = Relocation::getValue(Reloc.Type, TargetAddress, SectionAddress + Reloc.Offset);
```

Comment:
Can you run clang-format on this line? This line exceeds 80 columns.

Reply:
done, thank you :)